### ISSUE TITLE
Unable to Initialize a New Cache via the opencas.conf file

### ISSUE THREAD

**Issue by philam776** [Open-CAS/open-cas-linux#681]
## Question

I have previously successfully configured and ran a manual cache with an NVMe SSD as the caching device and a RAID 10 with 6 HDDs as the core device. However, I have configured the opencas.conf file in the way described in the guide and when running `casctl init` or `casctl init --force` nothing returns for me and a cache is neither initialized nor started when I run `casctl start`. When I am attempting this, the previous manually started cache is not running. `casctl stop` works fine in order to stop a running cache, but I am unable to successfully start a cache from the opencas.conf file. I attached my opencas.conf file below in case I was missing something.

Thank you
## opencas.conf file
```
version=19.3.0

## Caches configuration section
[caches]
## Cache ID    Cache device                Cache mode    Extra fields (optional)
1 /dev/disk/by-id/nvme-eui.334534304a6042640025385800000001    WT

## Core devices configuration
[cores]
## Cache ID    Core ID        Core device    Extra fields (optional)
1 1 /dev/disk/by-id/wwn-0x600605b00e53d8e027a717e73bc291f5

## To specify use of the IO Classification file, place content of the following line in the
## Caches configuration section under Extra fields (optional)
## ioclass_file=/etc/opencas/ioclass-config.csv

## If given cache/core pair is especially slow to start up, often doesn't come back
## up after reboot or you simply don't care if it does and don't want it to have
## an effect on your boot you can mark cores as lazy to prevent Open CAS from
## dropping boot to emergency shell because of this core failure. To do this
## put following line under in Extra fields (optional) section of core config:
## lazy_startup=true
## NOTE: This will cause open-cas.service to not wait for marked core while
## starting up - this option should be used with care to prevent races with
## other services for devices (e.g. mounts based on FS labels)
```

**Comment by Ostrokrzew** [comment]
Hi, could You paste also output of `lsblk` and `ls -l /dev/disk/by-id/`? It would be helpful. And please mark devices You want to use with CAS.

**Comment by philam776** [comment]
Absolutely. I am using nvme0n1 as my cache device and sdb as my core device.
`lsblk`
```
NAME    MAJ:MIN RM   SIZE RO TYPE MOUNTPOINT 
loop0     7:0    0  97.9M  1 loop /snap/core/10583
loop1     7:1    0  62.1M  1 loop /snap/gtk-common-themes/1506
loop3     7:3    0  49.8M  1 loop /snap/snap-store/467
loop4     7:4    0  29.9M  1 loop /snap/snapd/8542
loop5     7:5    0  31.1M  1 loop /snap/snapd/10707
loop6     7:6    0  64.8M  1 loop /snap/gtk-common-themes/1514
loop7     7:7    0    51M  1 loop /snap/snap-store/518
loop8     7:8    0 255.6M  1 loop /snap/gnome-3-34-1804/36
loop9     7:9    0   219M  1 loop /snap/gnome-3-34-1804/66
loop10    7:10   0   9.1M  1 loop /snap/canonical-livepatch/95
loop11    7:11   0  55.4M  1 loop /snap/core18/1944
loop12    7:12   0  55.5M  1 loop /snap/core18/1988
sda       8:0    0   1.8T  0 disk 
├─sda1    8:1    0   512M  0 part /boot/efi
└─sda2    8:2    0   1.8T  0 part /
sdb       8:16   0  21.9T  0 disk 
nvme0n1 259:0    0 894.3G  0 disk 
```
`ls -l /dev/disk/by-id/`
```
total 0 
lrwxrwxrwx 1 root root 13 Feb  8 07:31 nvme-eui.334534304a6042640025385800000001 -> ../../nvme0n1
lrwxrwxrwx 1 root root 13 Feb  8 07:31 nvme-SAMSUNG_MZ1LW960HMJP-000MV_S3E4NX0J604264 -> ../../nvme0n1
lrwxrwxrwx 1 root root  9 Feb  8 07:31 scsi-3600605b00e53d8e027a6ee420b4c6ce4 -> ../../sda
lrwxrwxrwx 1 root root 10 Feb  8 07:31 scsi-3600605b00e53d8e027a6ee420b4c6ce4-part1 -> ../../sda1
lrwxrwxrwx 1 root root 10 Feb  8 07:31 scsi-3600605b00e53d8e027a6ee420b4c6ce4-part2 -> ../../sda2
lrwxrwxrwx 1 root root  9 Feb  8 07:31 scsi-3600605b00e53d8e027a717e73bc291f5 -> ../../sdb
lrwxrwxrwx 1 root root  9 Feb  8 07:31 wwn-0x600605b00e53d8e027a6ee420b4c6ce4 -> ../../sda
lrwxrwxrwx 1 root root 10 Feb  8 07:31 wwn-0x600605b00e53d8e027a6ee420b4c6ce4-part1 -> ../../sda1
lrwxrwxrwx 1 root root 10 Feb  8 07:31 wwn-0x600605b00e53d8e027a6ee420b4c6ce4-part2 -> ../../sda2
lrwxrwxrwx 1 root root  9 Feb  8 07:31 wwn-0x600605b00e53d8e027a717e73bc291f5 -> ../../sdb
```

**Comment by Ostrokrzew** [comment]
Can You show `dmesg` logs from `casctl init --force`? And are there any messages from CAS after command fails? I need also version of CAS (`casadm -V`) and, if it's possible, Your's system distribution and version (e.g. `hostnamectl` output).

**Comment by philam776** [comment]
I was unable to find anything from `casctl init --force` when running `dmesg`. There's no output or error messages from CAS after the command fails, it just returns immediately to the next line. `casadm -V` output is shown below. My distro and version is Ubuntu 20.04.2 LTS.

╔═════════════════════════╤═════════════════════╗ 
║ Name                    │       Version       ║
╠═════════════════════════╪═════════════════════╣
║ CAS Cache Kernel Module │ 20.12.0.0438.master ║
║ CAS Disk Kernel Module  │ 20.12.0.0438.master ║
║ CAS CLI Utility         │ 20.12.0.0438.master ║
╚═════════════════════════╧═════════════════════╝

**Comment by Ostrokrzew** [comment]
I have to ask You one more question. Which `opencas.conf` file You have modified? Open CAS uses config file placed in `/etc/opencas/opencas.conf`. It's copied there during installation. Also `exit code` of `casctl init` could tell us something.

**Comment by philam776** [comment]
I have modified the `opencas.conf` file in `etc/open-cas-linux-20.12.0.0438.master/utils/opencas.conf`. that is where it was originally downloaded. I did not see a copy of that `opencas.conf` file in the main `/open-cas-linux-20.12.0.0438.master`. I tried to copy and paste that `opencas.conf` file into the main folder, then ran castctl --force and that had the same outcome as I have spoken about above.

**Comment by Ostrokrzew** [comment]
How did You install Open CAS?
Please, look at line 141 in Makefile (in *casadm* directory):
```
@install -m 644 $(UTILS_DIR)/opencas.conf $(DESTDIR)/etc/opencas/opencas.conf
```
`DESTDIR` is used during installation if You install Open CAS as a package, otherwise it's empty.

**Comment by philam776** [comment]
I installed Open CAS with:
`wget https://github.com/Open-CAS/open-cas-linux/releases/download/v20.12.1/open-cas-linux-20.12.1.0442.release.tar.gz
tar -xf open-cas-linux-20.12.1.0442.release.tar.gz
cd open-cas-linux-20.12.1.0442.release/`

This is my line 141 in Makefile:
`@install -m 644 $(UTILS_DIR)/opencas.conf $(DESTDIR)/etc/opencas/opencas.conf`

**Comment by Ostrokrzew** [comment]
What You wrote about installation process looks incomplete. Here are some step-by-step instructions for You and everybody.

1. It is preferred to unzip the package into Your home directory (or some non-global directory);
2. `cd` to this directory;
3. call `./configure`;
4. call `make`
5. call `make install`
6. call `casadm -V` to check if Open CAS is installed and modules are loaded;
7. edit config placed in `/etc/opencas/opencas.conf` - it's an absolute path;
8. call `casctl init (--force)`;
9. check if Open CAS is running.

And there is our manual: https://open-cas.github.io/getting_started_open_cas_linux.html
It is applicable for You and the only difference is that You downloaded tarball, not cloned repository so you already have submodules.

**Comment by Ostrokrzew** [comment]
PS You said You edited opencas.conf in source files. You may not have to edit it in `/etc/opencas/opencas.conf` but it's better to check twice and read it.

**Comment by Ostrokrzew** [comment]
@philam776 Has it helped?

**Comment by philam776** [comment]
Yes, I did a `make uninstall` and installed it again with those same steps as you listed above. But, that opencas.conf file wasn't originally in my `/etc/opencas` directory. I copied the file in there before I did a `make install` again then I started it with `casadm init --force` and it was able to find that pathway and start the cache. OpenCAS starts on a restart of the system as well. Thank you for your help.
